### PR TITLE
C++ fix rotation serialization

### LINKED ISSUE #3865: C++: Failure to serialize rotations (`code-examples/transform3d_simple`)
Repro: `./docs/code-examples/roundtrips.py transform3d_simple`

This might be related to why `just cpp-test` fails

### PR DESCRIPTION
**Pull request by Wumpf**
### What

* Fixes #3865

It seems this was caused by undefined behavior via calling an assignment operator on uninitialized memory!

Bulldozered the entire responsible codegen section over to make it super simple :)
(which we can do now in part because recent std::array use - should have done so from the start, makes things way simpler!)

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested [demo.rerun.io](https://demo.rerun.io/pr/3935) (if applicable)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG

- [PR Build Summary](https://build.rerun.io/pr/3935)
- [Docs preview](https://rerun.io/preview/b0a17e547eb59de9f2be8982e8ae1d01ca15afb5/docs) 
- [Examples preview](https://rerun.io/preview/b0a17e547eb59de9f2be8982e8ae1d01ca15afb5/examples) 
- [Recent benchmark results](https://ref.rerun.io/dev/bench/)
- [Wasm size tracking](https://ref.rerun.io/dev/sizes/)

### CODE REVIEW

**Comment by Wumpf** [comment]
oh and as expected this also fixes issues for me that I saw on `just cpp-test` but not on CI (reportedly Emil experienced the same)!

File: rerun_cpp/tests/generated/datatypes/affix_fuzzer4.hpp [hunk 1]
```diff
@@ -120,30 +120,31 @@ namespace rerun {
...
             static AffixFuzzer4 many_optional(
                 std::optional<std::vector<rerun::datatypes::AffixFuzzer3>> many_optional
             ) {
-                using TypeAlias = std::optional<std::vector<rerun::datatypes::AffixFuzzer3>>;
                 AffixFuzzer4 self;
                 self._tag = detail::AffixFuzzer4Tag::many_optional;
-                new (&self._data.many_optional) TypeAlias(std::move(many_optional));
+                new (&self._data.many_optional
+                ) std::optional<std::vector<rerun::datatypes::AffixFuzzer3>>(std::move(many_optional
+                ));
```
Comment (emilk) [inline rerun_cpp/tests/generated/datatypes/affix_fuzzer4.hpp:147]: this formatting 😬

**Review by emilk — APPROVED** [review]
Super!

File: rerun_cpp/tests/generated/datatypes/affix_fuzzer4.hpp (hunk 1 above)
Comment (Wumpf) [reply rerun_cpp/tests/generated/datatypes/affix_fuzzer4.hpp:147]: it's just a test and it wasn't me, it was clang format :D ;)

**Pull request merged**